### ISSUE TITLE
Import of pre-WP4.1 images breaks URLs due to change of image scaling/rounding

### ISSUE THREAD

**Issue by kraftner** [humanmade/WordPress-Importer#64]
[#30660](https://core.trac.wordpress.org/changeset/30660) changed the scaling behaviour for attachments. Due to that when importing a WXR exported from pre 4.1 and then importing it in a recent version when generating thumbs different sizes/urls get created with the same settings, in fact ones which might be off by one pixel. Therefore when those images are used in posts they are broken after import.

For reference I am adding an extensive [WPSE question](http://wordpress.stackexchange.com/q/202622/47733) as well as the [corresponding ticket #31581 in trac](https://core.trac.wordpress.org/ticket/31581)

## What is expected?

When importing posts with image in it URLs are updated so the post still shows the image.

## What is happening

Those images that get scaled to a different size post 4.1 break as their size and therefore filename might be off by 1 Pixel.

## Solutions?

I have created [a workaround for the old importer](http://wordpress.stackexchange.com/a/206992/47733) that roughly does this:

1. During the import it hooks into `wp_update_attachment_metadata` and calculates the scaled image size for all imported images according to the old scaling behaviour.
2. If there is a mismatch the offending URL is added to an array…
3. …which is then at the end used to go through all posts to replace those urls.

The way this is implemented is a bit convoluted due to the way the old importer works. Fact is that there already is one

**Comment by kraftner** [comment]
Oh I have just realized that [for now you are totally ignoring resized images](https://github.com/humanmade/WordPress-Importer/blob/0ffae291bfa4e77268e6f7f46aef9a5be8a0992c/class-wxr-importer.php#L1067-L1078) leaving the original URLs in. Then I am leaving the above as a note to keep that in mind when building that feature at some point

**Comment by rmccue** [comment]
Related to #53 I think.

**Comment by kraftner** [comment]
Yes kind of. Once resized images in `post_content` get replaced as well as requested in #53 the issue I described above will come become an issue and needs to be considered.

**Comment by bobbingwide** [comment]
+1

**Comment by mixcas** [comment]
+1
